File: scripts/test_monetization.js
#!/usr/bin/env node

/**
 * ASF App - Monetization Test Suite
 * Tests all monetization features without external dependencies
 */

const fs = require('fs');
const path = require('path');

// Test counters
let passed = 0;
let failed = 0;

function test(name, condition) {
  if (condition) {
    console.log(`✅ ${name}`);
    passed++;
  } else {
    console.log(`❌ ${name}`);
    failed++;
  }
}

// Read index.html
const html = fs.readFileSync(path.join(__dirname, '..', 'index.html'), 'utf8');

console.log('\n🧪 ASF Monetization Test Suite\n');
console.log('=' .repeat(50));

// 1. Structure Tests
console.log('\n📦 Structure Tests:\n');

test('Has Loja section', html.includes('id="loja"'));
test('Has affiliate-store container', html.includes('id="affiliate-store"'));
test('Has premium-modal', html.includes('id="premium-modal"'));
test('Has newsletter-email input', html.includes('id="newsletter-email"'));
test('Has WhatsApp button', html.includes('contactWhatsApp'));
test('Has showPremiumModal function', html.includes('function showPremiumModal()'));
test('Has hidePremiumModal function', html.includes('function hidePremiumModal()'));
test('Has subscribeNewsletter function', html.includes('function subscribeNewsletter()'));
test('Has openAffiliateLink function', html.includes('function openAffiliateLink('));
test('Has renderAffiliateStore function', html.includes('function renderAffiliateStore()'));

// 2. Products Tests
console.log('\n🛒 Products Tests:\n');

test('Has AFFILIATE_PRODUCTS array', html.includes('const AFFILIATE_PRODUCTS = ['));
test('Product: Prancha Soft Caril', html.includes('prancha-soft-caril'));
test('Product: Lycra DryFit', html.includes('lycra-termica-dryfit'));
test('Product: Wax FCS', html.includes('wax-fcs'));
test('Product: Poncho Microfibra', html.includes('poncho-microfibra'));
test('Mercado Livre URLs present', html.includes('mercadolivre.com.br'));
test('6 products in array', (html.match(/id:/g) || []).length >= 6);

// 3. Navigation Tests
console.log('\n🧭 Navigation Tests:\n');

test('Loja nav button exists', html.includes('aria-label="Loja e ofertas"'));
test('Premium nav button exists', html.includes('aria-label="Tornar-se Premium"'));
test('Loja nav has correct onclick', html.includes("handleNavTap(this, 'loja')"));
test('Premium nav calls showPremiumModal', html.includes("onclick=\"showPremiumModal()\""));

// 4. Tracking Tests
console.log('\n📊 Analytics Tests:\n');

test('GA4 script present', html.includes("gtag('config', 'G-"));
test('trackEvent function', html.includes('function trackEvent('));
test('getMonetizationStats function', html.includes('function getStats()'));
test('ASFMonetization global object', html.includes('window.ASFMonetization'));
test('LocalStorage tracking key', html.includes("'asf-monetization-stats'"));
test('Affiliate clicks tracking', html.includes("'asf-affiliate-clicks'"));

// 5. GA4 Events
console.log('\n🎯 GA4 Event Tests:\n');

test('affiliate_click event', html.includes("'affiliate_click'"));
test('newsletter_signup event', html.includes("'newsletter_signup'"));
test('premium_modal_view event', html.includes("'premium_modal_view'"));
test('payment_start event', html.includes("'payment_start'"));
test('contact_whatsapp event', html.includes("'contact_whatsapp'"));

// 6. UI/UX Tests
console.log('\n🎨 UI/UX Tests:\n');

test('Toast notification function', html.includes('function showToast('));
test('Toast animation CSS', html.includes('@keyframes toastIn'));
test('Modal animation CSS', html.includes('@keyframes modalBounce'));
test('Premium CTA banner', html.includes('Apoie a ASF - Vire Premium'));
test('Newsletter section', html.includes('id="monetization-cta"'));
test('How it works explanation', html.includes('Como funciona'));

// 7. Content Tests
console.log('\n📝 Content Tests:\n');

test('Botão "Ver oferta" text', html.includes('🛒 Ver oferta'));
test('Premium price displayed', html.includes('R$ 19,90'));
test('7 days free mention', html.includes('7 dias grátis'));
test('Mercado Pago mention', html.includes('Mercado Pago'));
test('WhatsApp number correct', html.includes('5511954346288'));
test('Email privacy notice', html.includes('seu email está seguro'));

// 8. Responsive & Mobile
console.log('\n📱 Responsive Tests:\n');

test('Viewport meta', html.includes('viewport'));
test('Outfit font family', html.includes("'Outfit'"));
test('CSS custom properties', html.includes('--primary'));
test('Grid layout for products', html.includes('grid-template-columns'));

// Summary
console.log('\n' + '=' .repeat(50));
console.log(`\n📊 Test Results: ${passed} passed, ${failed} failed`);

if (failed === 0) {
  console.log('\n🎉 All tests passed! Monetization is ready.\n');
} else {
  console.log('\n⚠️  Some features may need attention.\n');
  process.exit(1);
}

// Additional info
console.log('💡 Next steps:');
console.log('   1. Get Google AdSense approval (ca-pub-XXXX)');
console.log('   2. Set up Mercado Pago API keys');
console.log('   3. Configure MailerLite API');
console.log('   4. Test manually: https://acarolmourad-commits.github.io/asf-app/');
console.log('   5. Verify GA4 real-time events\n');

File: scripts/test_monetization.py
#!/usr/bin/env python3
"""ASF Monetization Test Suite - validates HTML structure"""

import re, sys

with open('index.html', 'r', encoding='utf-8') as f:
    html = f.read()

passed = 0
failed = 0

def test(name, condition):
    global passed, failed
    if condition:
        print(f'✅ {name}')
        passed += 1
    else:
        print(f'❌ {name}')
        failed += 1

print('\n🧪 ASF Monetization Test Suite\n' + '='*50)

# Structure
print('\n📦 Structure:\n')
test('Loja section exists', 'id="loja"' in html)
test('affiliate-store container', 'id="affiliate-store"' in html)
test('premium-modal exists', 'id="premium-modal"' in html)
test('newsletter-email input', 'id="newsletter-email"' in html)
test('WhatsApp contact func', 'function contactWhatsApp()' in html)
test('showPremiumModal func', 'function showPremiumModal()' in html)
test('subscribeNewsletter func', 'function subscribeNewsletter()' in html)
test('openAffiliateLink func', 'function openAffiliateLink(' in html)
test('renderAffiliateStore func', 'function renderAffiliateStore()' in html)

# Products
print('\n🛒 Products:\n')
test('AFFILIATE_PRODUCTS array', 'const AFFILIATE_PRODUCTS = [' in html)
test('Product: Prancha Soft Caril', 'prancha-soft-caril' in html)
test('Product: Lycra DryFit', 'lycra-termica-dryfit' in html)
test('Product: Wax FCS', 'wax-fcs' in html)
test('Product: Poncho', 'poncho-microfibra' in html)
test('Mercado Livre URLs', 'mercadolivre.com.br' in html)
test('6+ products', html.count("id: '") >= 6)

# Navigation
print('\n🧭 Navigation:\n')
test('Loja nav button', 'aria-label="Loja e ofertas"' in html)
test('Premium nav button', 'aria-label="Tornar-se Premium"' in html)
test('Loja onclick correct', "handleNavTap(this, 'loja')" in html)
test('Premium onclick modal', 'showPremiumModal()' in html)

# Tracking
print('\n📊 Analytics:\n')
test('GA4 script present', "gtag('config', 'G-" in html)
test('trackEvent function', 'function trackEvent(' in html)
test('getStats function', 'function getStats()' in html)
test('ASFMonetization object', 'window.ASFMonetization' in html)
test('monetization-stats key', "'asf-monetization-stats'" in html)
test('affiliate-clicks key', "'asf-affiliate-clicks'" in html)

# GA4 Events
print('\n🎯 Events:\n')
test('affiliate_click event', "'affiliate_click'" in html)
test('newsletter_signup event', "'newsletter_signup'" in html)
test('premium_modal_view event', "'premium_modal_view'" in html)
test('payment_start event', "'payment_start'" in html)

# UI/UX
print('\n🎨 UI:\n')
test('Toast notifications', 'function showToast(' in html)
test('Modal animation', '@keyframes modalBounce' in html)
test('Toast animation', '@keyframes toastIn' in html)
test('Premium CTA banner', 'Apoie a ASF - Vire Premium' in html)
test('Newsletter section', 'id="monetization-cta"' in html)

# Content
print('\n📝 Content:\n')
test('"Ver oferta" button', '🛒 Ver oferta' in html)
test('Premium price R$19,90', 'R$ 19,90' in html)
test('7 days free text', '7 dias grátis' in html)
test('Mercado Pago', 'Mercado Pago' in html)
test('WhatsApp number', '5511954346288' in html)
test('Privacy notice', 'seu email está seguro' in html)

print('\n' + '='*50)
print(f'\n📊 Results: {passed} passed, {failed} failed')

if failed == 0:
    print('\n🎉 All tests passed! Monetization is ready.')
    print('\n💡 Action items:')
    print('   1. Get Google AdSense approval')
    print('   2. Set up Mercado Pago API')
    print('   3. Configure MailerLite API key')
    print('   4. Test manually: https://acarolmourad-commits.github.io/asf-app/')
    print('   5. Verify GA4 real-time events\n')
    sys.exit(0)
else:
    print('\n⚠️  Some checks failed - review implementation.\n')
    sys.exit(1)
